infra.bs

```diff
@@ -174,6 +174,78 @@ caller's algorithm. Using prose the caller has the ability to "catch" the except
 another action.
 
 
+<h3 id=algorithm-conditional-abort>Conditional abort</h3>
+
+<p>Sometimes it is useful to stop performing a series of steps once some condition becomes true.
+
+<p>To do this, state that a given series of steps will <dfn export>abort when</dfn> a specific
+<var>condition</var> is reached. This indicates that the specified steps must be evaluated, not
+as-written, but by additionally inserting a step before each of them that evaluates
+<var>condition</var>, and if <var>condition</var> evaluates to true, skips the remaining steps.
```

Comment:
Do we need to make it clear that this is "deep"? As in, will be evaluated before each sub-step too.

Reply:
Since we're trying to move away from substeps and haven't really discussed nested steps thus far I've not done this. I think that's okay for now, but welcome further discussion.